server/issue.go

```diff
@@ -1096,30 +1096,34 @@ func (p *Plugin) GetIssueByKey(instanceID, mattermostUserID types.ID, issueKey s
 	return issue, nil
 }
 
-func (p *Plugin) GetIssueDataWithAPIToken(issueID string, instanceID string) (*jira.Issue, error) {
+func (p *Plugin) GetIssueDataWithAPIToken(issueID, instanceID string) (*jira.Issue, error) {
 	client := &http.Client{}
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s/rest/api/2/issue/%s", instanceID, issueID), nil)
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/rest/api/2/issue/%s", instanceID, issueID), nil)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to create http request for fetching issue data. IssueID: %s", issueID)
 	}
 
 	req.Header.Set("Authorization", fmt.Sprintf("Basic %s", p.getConfig().AdminAPIToken))
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to get issue data. IssueID: %s", issueID)
+	}
+
+	if resp.Body == nil {
+		return nil, errors.Wrapf(err, "failed to get issue data. IssueID: %s", issueID)
```

Comment:
The API was successful in this case, I think we can write something like `missing data for issue` or `no data exist for issue`. We can show status code like below